web/src/utils/localstorage.ts

```diff
@@ -0,0 +1,28 @@
+import { Signal, createEffect, createRoot, createSignal } from 'solid-js';
+
+function createStorageSignal(
+  storage: Storage,
+  name: string,
+): Signal<string | undefined> {
+  const [value, setValue] = createSignal<string | undefined>(
+    storage.getItem(name) || undefined,
+  );
+  // The session writer effect's lifecycle should be tied to that of the signal
+  createRoot(() => {
+    createEffect(() => {
```

Comment:
Should we re-validate this value after every render call? Might be a little excessive.

Reply:
We don't, the effect is only updated when `value()` changes.

Reply:
Okay then I misinterpreted the function doc comment: `Creates a reactive computation that runs after the render phase`.

Reply:
It runs after the render phase, if it has a reason to :P